Obstacles

guilbill opened the pull request:
This PR aim to add obstacles to the board. Trello : https://trello.com/c/ndwsjSNg/2-05-as-pierre-i-need-obstacles-because-only-moving-the-puck-on-a-square-board-doesnt-teach-me-a-thing
![tipsy-obstacles](https://user-images.githubusercontent.com/560852/105157272-18943280-5b0d-11eb-91e7-1e5282085d0e.gif)

guillbill committed:
chore(refactor): Move display logic in new game class

guillbill committed:
feat(instruction): add intruction for directions

guillbill committed:
feat(obstacles): add obstacles

guillbill committed:
chore(clean): remove debug print

guillbill committed:
feat(display): clean screen before all

guillbill committed:
chore(clean): remove debug print again...

fzaninotto commented on src/board.py:
```diff
@@ -28,27 +35,15 @@ def __initialize_empty_board(self):
     def __get_next_free_node(self, node, direction):
         next_node = [next_node for start_node, next_node, edge_attrs in self.graph.out_edges(
-            node, data=True) if edge_attrs.get('direction') == direction]
+            node, data=True) if (edge_attrs.get('direction') == direction and not self.graph.nodes[next_node].get('obstacle'))]
```
you should remove the edges to obstacles instead

guilbill replied:
done

fzaninotto replied:
should be removed

guilbill replied:
This time it's really gone

fzaninotto commented on src/game.py:
```diff
@@ -0,0 +1,57 @@
+
+            self.board.tilt(input_command)
+
+    def display_board(self):
```
You should unit test this method at least once

guilbill replied:
You mean unit test the display of the board?

fzaninotto replied:
yes. That will force you to separate string formatting and string output.

guilbill replied:
done

fzaninotto commented on src/game.py:
```diff
@@ -0,0 +1,57 @@
+            print(line)
+        print('# '*(self.board.WIDTH+2))
+
+    def display_title(self):
```
👍 badass

zyhou commented on src/game.py:
```diff
@@ -0,0 +1,57 @@
+        print("")
+        input("Welcome to Tipsy game. Type Enter to start")
+
+    def display_instructions(self):
```
:+1:

guillbill committed:
chore(obstacle): obstacles are materialiwed by an absence of node in the graph.

Instead of adding an attribute "obstacle" on the node, it's smartest to remove the node from the graph.

zyhou commented on src/board.py:
```diff
@@ -18,8 +18,7 @@ def __initialize_pucks_positions(self):
     def __initialize_obstacles_positions(self, obstacles):
-        for obstacle in obstacles:
-            self.graph.nodes[obstacle]["obstacle"] = True
+        self.graph.remove_nodes_from(obstacles)
```
Nice update

guillbill committed:
test(display): Test display of board

guillbill committed:
fix(obstacle): remove wrong condition

fzaninotto reviewed (approved):
Perfect!

guillbill committed:
chore(travis): add travis config file

guillbill committed:
refacto(board): rename some variables

guillbill committed:
fix(testGame name): rename TeatGame correctly
